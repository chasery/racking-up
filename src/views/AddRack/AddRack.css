.AddRack {
  padding-top: 3.75rem;
}

.AddRack__wrapper {
  display: flex;
  justify-content: center;
  padding: 1rem 1.5rem;
}

@media all and (min-width: 50rem) {
  .AddRack__wrapper {
    max-width: 64rem;
    align-self: center;
    margin: 0 auto;
  }
}
