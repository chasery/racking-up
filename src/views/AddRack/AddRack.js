import React, { useContext, useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import RacksContext from '../../contexts/RacksContext';
import RacksApiService from '../../services/racks-api-service';
import Form from '../../components/Form/Form';
import FormField from '../../components/FormField/FormField';
import './AddRack.css';

function AddRack(props) {
  const context = useContext(RacksContext);
  const [name, setName] = useState('');
  const history = useHistory();

  useEffect(() => {
    context.clearError();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleAddRack = (e) => {
    e.preventDefault();
    RacksApiService.postRack({
      rack_name: name,
    })
      .then((res) => {
        context.addRack(res);
        history.push(`/racks/${res.rack_id}`);
      })
      .catch(context.setErrorState);
  };

  return (
    <main role='main'>
      <section className='AddRack'>
        <div className='AddRack__wrapper'>
          <Form id='AddRack' onSubmit={(e) => handleAddRack(e)}>
            <h2>Add Rack</h2>
            <p>Enter the name of the rack you would like to add.</p>
            <FormField
              id='name'
              label='Name'
              type='text'
              isRequired={true}
              onChange={(e) => setName(e.target.value)}
              value={name}
            />
            <button onClick={() => history.goBack()}>Cancel</button>
            <button type='submit'>Add Rack</button>
          </Form>
        </div>
      </section>
    </main>
  );
}

export default AddRack;
