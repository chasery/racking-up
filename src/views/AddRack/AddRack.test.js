import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddRack from "./AddRack";

describe("AddRack component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<AddRack />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
