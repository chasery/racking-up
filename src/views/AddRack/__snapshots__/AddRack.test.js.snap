// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRack component renders without error 1`] = `
<main
  role="main"
>
  <section
    className="AddRack"
  >
    <div
      className="AddRack__wrapper"
    >
      <Form
        id="AddRack"
        onSubmit={[Function]}
      >
        <h2>
          Add Rack
        </h2>
        <p>
          Enter the name of the rack you would like to add.
        </p>
        <FormField
          id="name"
          isRequired={true}
          label="Name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          Cancel
        </button>
        <button
          type="submit"
        >
          Add Rack
        </button>
      </Form>
    </div>
  </section>
</main>
`;
